src/Sql/Sql/Elastic Jobs/Cmdlet/JobStep/SetAzureSqlElasticJobStep.cs

```diff
@@ -0,0 +1,532 @@
+        [ValidateNotNullOrEmpty]
+        public override string ServerName { get; set; }
+
+        /// <summary>
+        /// Gets or sets the agent name
+        /// </summary>
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = DefaultParameterSet,
+            Position = 2,
+            HelpMessage = "The agent name")]
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = DefaultRemoveOutputParameterSet,
+            Position = 2,
+            HelpMessage = "The agent name")]
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = DefaultAddParentResourceIdParameterSet,
+            Position = 2,
+            HelpMessage = "The agent name")]
+        [ValidateNotNullOrEmpty]
+        public override string AgentName { get; set; }
+
+        /// <summary>
+        /// Gets or sets the job name
+        /// </summary>
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = DefaultParameterSet,
+            Position = 3,
+            HelpMessage = "The job name")]
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = DefaultRemoveOutputParameterSet,
+            Position = 3,
+            HelpMessage = "The job name")]
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = DefaultAddParentResourceIdParameterSet,
+            Position = 3,
+            HelpMessage = "The job name")]
+        [ValidateNotNullOrEmpty]
+        public override string JobName { get; set; }
+
+        /// <summary>
+        /// Gets or sets the job step name
+        /// </summary>
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = DefaultParameterSet,
+            HelpMessage = "The step name")]
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = DefaultRemoveOutputParameterSet,
+            HelpMessage = "The step name")]
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = DefaultAddParentResourceIdParameterSet,
+            HelpMessage = "The step name")]
+        [Alias("StepName")]
+        [ValidateNotNullOrEmpty]
+        public override string Name { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = DefaultRemoveOutputParameterSet,
+            HelpMessage = "The flag to indicate whether to remove output")]
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = ParentObjectRemoveOutputParameterSet,
+            Position = 1,
+            HelpMessage = "The flag to indicate whether to remove output")]
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = ParentResourceIdRemoveOutputParameterSet,
+            Position = 1,
+            HelpMessage = "The flag to indicate whether to remove output")]
+        public SwitchParameter RemoveOutput { get; set; }
```

Comment:
@johnpaulkee what is this parameter supposed to do?

Reply:
Job steps have an optional feature to have the output of a T-Sql query from a successfully executed step to go to a target output server's db on a specific table.

This switch removes the output for a step if a customer would like to do so.